src/components/Navbar/index.tsx

```diff
@@ -220,16 +216,16 @@ export class NavbarBase extends React.Component<Props, State> {
           </div>
         </Navbar.Brand>
         <Navbar.Text className={styles.text}>
-          {currentVersion && file ? (
+          {currentVersionId ? (
             <a
               className={styles.legacyLink}
               href={
-                baseFileId
-                  ? `${baseUrlToLegacy}/compare/${file.id}...${baseFileId}/${
-                      path ? `file/${path}` : ''
+                currentBaseVersionId
+                  ? `${baseUrlToLegacy}/compare-redirect/${currentVersionId}...${currentBaseVersionId}/${
+                      path ? `?file=${path}` : ''
```

Comment:
Would having a helper function to create this query string (taking `path` as unique argument) make sense?

Reply:
It seems like we'd be adding more lines of code to define (and call) a function that does this simple calculation, than we currently have, but you're right, it is duplicated logic, so I guess it makes sense. I'll add one. I don't think we want to go the next step and use a mock for that function in tests, and have a separate test for the function though, would we? That seems like creating a lot of extra code and work for very little benefit.

Reply:
No, no mock. We can reuse the helper function in the test, though.